superset-frontend/packages/superset-core/src/api/sqlLab.ts

```diff
@@ -111,6 +111,171 @@ export interface Tab {
   panels: Panel[];
 }
 
+/**
+ * Generic data types, see enum of the same name in superset/utils/core.py.
+ */
+enum GenericDataType {
+  Numeric = 0,
+  String = 1,
+  Temporal = 2,
+  Boolean = 3,
+}
+
+/**
+ * Column metadata returned in query results.
+ */
+export type QueryColumn = {
+  /**
+   * Label of the column
+   */
+  name?: string;
+
+  /**
+   * Column name defined
+   */
+  column_name: string;
+
+  /**
+   * Type of the column value
+   */
+  type: string | null;
+
+  /**
+   * Generic data type format
+   */
+  type_generic: GenericDataType;
+
+  /**
+   * True if the column is date format
+   */
+  is_dttm: boolean;
+};
+
+export enum CTASMethod {
+  Table = 'TABLE',
+  View = 'VIEW',
+}
+
+export interface CTAS {
+  /**
+   * Create method for CTAS creation request
+   */
+  method: CTASMethod;
+
+  /**
+   * Temporary table name for creation using a CTAS query
+   */
+  tempTable: string | null;
+}
+
+export interface QueryRequestContext {
```

Comment:
Given that the properties defined here are used both when requesting a query but also for query results and errors, I think we should remove the `Request` word from the name and just call it `QueryContext`.

Reply:
@villebro I think this concept is defined on the backend but we don't use in the frontend. I'll add a topic for our meeting to discuss this as well.